geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonPathParser.scala

```diff
@@ -122,33 +119,22 @@ private class JsonPathParser extends Parser {
   private def pushDeepScan(context: Context[Any]): Unit =
     context.getValueStack.push(PathDeepScan :: context.getValueStack.pop.asInstanceOf[List[_]])
 
-  def ArrayIndex: Rule1[PathIndex] = rule { "[" ~ Number ~ "]" ~~> PathIndex }
+  def ArrayIndex: Rule1[PathIndex] = rule { "[" ~ int ~ "]" ~~> PathIndex }
 
   def ArrayIndices: Rule1[PathIndices] =
-    rule { "[" ~ Number ~ "," ~ oneOrMore(Number, ",") ~ "]" ~~> ((n0, n) => PathIndices(n.+:(n0))) }
+    rule { "[" ~ int ~ "," ~ oneOrMore(int, ",") ~ "]" ~~> ((n0, n) => PathIndices(n.+:(n0))) }
 
   def ArrayIndexRange: Rule1[PathIndices] =
-    rule { "[" ~ Number ~ ":" ~ Number ~ "]" ~~> ((n0, n1) => PathIndices(n0 until n1)) }
+    rule { "[" ~ int ~ ":" ~ int ~ "]" ~~> ((n0, n1) => PathIndices(n0 until n1)) }
+
+  def Attribute: Rule1[PathAttribute] = rule { "." ~ oneOrMore(char) ~> { (s) => PathAttribute(s) } ~ !"()" }
 
-  def Attribute: Rule1[PathAttribute] = rule { "." ~ oneOrMore(Character) ~> PathAttribute ~ !"()" }
+  def BracketedAttribute: Rule1[PathAttribute] =
+    rule { "[" ~ bracketedString ~~> { (s) => PathAttribute(s, bracketed = true) } ~ "]" }
 
-  def BracketedAttribute: Rule1[BracketedPathAttribute] = rule { ".['" ~ oneOrMore(CharacterWithDelimiter) ~> BracketedPathAttribute ~ !"()" ~ "']" }
+  private def bracketedString: Rule1[String] = unquotedString | singleQuotedString
 
   def Function: Rule1[PathFunction] = rule {
     "." ~ ("min" | "max" | "avg" | "length") ~> ((f) => PathFunction(JsonPathFunction.withName(f))) ~ "()"
   }
-
-  def Number: Rule1[Int] = rule { oneOrMore("0" - "9") ~> (_.toInt) }
-
-  def Character: Rule0 = rule { EscapedChar | NormalChar }
-
-  def CharacterWithDelimiter: Rule0 = rule { Character | " " | "." }
-
-  def EscapedChar: Rule0 = rule { "\\" ~ (anyOf("\"\\/bfnrt ") | Unicode) }
-
-  def NormalChar: Rule0 = rule { "a" - "z" | "A" - "Z" | "0" - "9" }
-
-  def Unicode: Rule0 = rule { "u" ~ HexDigit ~ HexDigit ~ HexDigit ~ HexDigit }
-
-  def HexDigit: Rule0 = rule { "0" - "9" | "a" - "f" | "A" - "F" }
```

Comment:
Were these superfluous previously?

Reply:
i think there were used at the time, but since then we've added a base parser class that defines them all (plus more), so i just extended that instead